pkg/detector/detector.go

```diff
@@ -1103,9 +1103,15 @@ func (d *ResourceDetector) HandlePropagationPolicyDeletion(policyID string) erro
 		return err
 	}
 
-	cleanupMarksFunc := func(obj metav1.Object) {
+	cleanupMarksFunc := func(obj metav1.Object) bool {
```

Comment:
The root cause of this issue is that `cleanupMarksFuncs` mistakenly removed the labels it was not supposed to.
The reason is that `cleanupMarksFuncs` just have the label keys, but can't verify the values.

This patch forces the `cleanupMarksFuncs` to couple with business logic, in another world, it makes it harder to test it with a unit test.

Another approach is to feed enough info(label key and value) to the `cleanupMarksFuncs`, and let it focus on removing certain things.